rest/route/project_copy.go

```diff
@@ -0,0 +1,79 @@
+package route
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen/rest/data"
+	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/pkg/errors"
+)
+
+////////////////////////////////////////////////////////////////////////
+//
+// POST /rest/v2/projects/{project_id}/copy
+
+type projectCopyHandler struct {
+	oldProjectId string
+	newProjectId string
+	sc           data.Connector
+}
+
+func makeCopyProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectCopyHandler{
+		sc: sc,
+	}
+}
+
+func (p *projectCopyHandler) Factory() gimlet.RouteHandler {
+	return &projectCopyHandler{
+		sc: p.sc,
+	}
+}
+
+func (p *projectCopyHandler) Parse(ctx context.Context, r *http.Request) error {
+	p.oldProjectId = gimlet.GetVars(r)["project_id"]
+	p.newProjectId = r.FormValue("new_project")
+	return nil
+}
+
+func (p *projectCopyHandler) Run(ctx context.Context) gimlet.Responder {
+	projectToCopy, err := p.sc.FindProjectById(p.oldProjectId)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrapf(err, "Database error finding project '%s'", p.oldProjectId))
+	}
+
+	// verify project with new ID doesn't exist
+	_, err = p.sc.FindProjectById(p.newProjectId)
```

Comment:
The function that gets the project from the db returns no error if not found, so this won't work

Reply:
The model function returns no error, but the connector handles this case and returns a StatusNotFound if the project is nil.